File: DSAndML/statistics/measureOfCentralTendencyforPD.R
# Function to calculate expected value
expectedValue <- function(distribution) {
    Mean <- 0
    # Loop through each element of the distribution
    for (i in seq_along(distribution)) {
        Mean <- Mean + ((i) * distribution[i])  # i is the index (which starts at 1 in R)
    }
    return(Mean)
}


# Function to calculate variance value
variance <- function(distribution) {
    Mean <- expectedValue(distribution)
    Var <- 0
    # Loop through each element of the distribution
    for (i in seq_along(distribution)) {
        X_squared <- (i) ** 2
        Var <- Var  +  (X_squared * distribution[i])
    }
    return(Var - (Mean ** 2))
}

expectedValue_1 <- function(distribution) {
    Mean <- 0
    for (i in 1:length(distribution)) {
       Mean <- Mean + (i * distribution[i])
    }
    return(Mean)
}

# Define the distribution using numeric fractions
distribution <- c(1/36, 3/36, 5/36, 7/36, 9/36, 11/36)

# Calculate and print the expected value
expected_val <- expectedValue(distribution)
print(paste("Expected Value:", round(expected_val,2)))

# Calculate and print the variance value
variance_val <- variance(distribution)
print(paste("Variance Value:", round(variance_val,2)))

File: DSAndML/statistics/measureOfCentralTendencyforPD.py
from cmath import sqrt
from fractions import Fraction as fr

# Function to calculate expected value
def expectedvalue(distribution):
    Mean = 0
    for i in range(len(distribution)):
        Mean += (i + 1) * distribution[i]  # (i + 1) is the outcome, distribution[i] is the probability
    return Mean

# Function to calculate variance
def variance(distribution):
    Mean = expectedvalue(distribution)  # Get the mean (expected value)
    var = 0
    for i in range(len(distribution)):
        X_squared = (i + 1) ** 2  # Square of the outcome (i + 1)
        var += X_squared * distribution[i]  # Multiply by probability P(X)
    return var - Mean ** 2  # Variance = E(X^2) - (E(X))^2

# Define the distribution using fractions
distribution = [fr(1, 36), fr(3, 36), fr(5, 36), fr(7, 36), fr(9, 36), fr(11, 36)]

# Calculate the expected value
expected_val = expectedvalue(distribution)

# Calculate the variance value
var = variance(distribution)

# Print the result in both fraction and decimal form
print(f"Expected Value (Fraction): {expected_val}")
print(f"Expected Value (Decimal): {float(expected_val)}")

# Print the result in both fraction and decimal form
print(f"Variance (Fraction): {var}")
print(f"Variance (Decimal): {float(var)}")

# Print the result in both fraction and decimal form
print(f"Standard Deviation (Fraction): {sqrt(var)}")
print(f"Standard Deviation (Decimal): {sqrt(float(var))}")